site/docs/hosting/cloudflare-workers-deno.md

```diff
@@ -0,0 +1,124 @@
+# Hosting: Cloudflare Workers (Deno)
+
+[Cloudflare Workers](https://workers.cloudflare.com/) is a public serverless computing platform that offers a convenient and simple solution for running small workloads at the [edge](https://en.wikipedia.org/wiki/Edge_computing).
+
+This guide will take you through the process of hosting your Telegram bot on Cloudflare Workers.
+
+## Prerequisites
+
+To follow along, please make sure that you have a [Cloudflare account](https://dash.cloudflare.com/login) with your workers subdomain [configured](https://dash.cloudflare.com/?account=workers).
+
+## Setting Things Up
+
+Make sure you have [Deno](https://deno.land/) and [Denoflare](https://denoflare.dev/) installed.
+
+Create a new directory and create a new file `.denoflare` in that directory.
+Put the following contents in the file:
+
+```json
+{
+  "$schema": "https://raw.githubusercontent.com/skymethod/denoflare/v0.5.11/common/config.schema.json",
+  "scripts": {
+    "my-bot": {
+      "path": "bot.ts",
+      "localPort": 3030,
+      "bindings": {
+        "BOT_TOKEN": {
+          "value": "YOUR_BOT_TOKEN"
+        }
+      },
+      "workersDev": true
+    }
+  },
+  "profiles": {
+    "account1": {
+      "accountId": "YOUR_ACCOUNT_ID",
+      "apiToken": "YOUR_API_TOKEN"
+    }
+  }
+}
+```
+
+Make sure to replace `YOUR_ACCOUNT_ID`, `YOUR_API_TOKEN` and `YOUR_BOT_TOKEN` appropriately.
+When creating your API token you can chose the `Edit Cloudflare Workers` preset from the pre-configured permissions.
+
+## Creating Your Bot
+
+Create a new file named `bot.ts` and put the following contents in it:
+
+```ts
+import {
+  Bot,
+  webhookCallback,
+} from "https://deno.land/x/grammy@v1.15.3/mod.ts";
+
+interface Environment {
+  BOT_TOKEN: string;
+}
+
+export default {
+  fetch(request: Request, env: Environment) {
+    try {
+      const bot = new Bot(env.BOT_TOKEN);
```

Comment:
Recreating the bot instance per request is problematic because the initialisation will have to happen again and again. Given that CFW has quite an ugly API with the env vars not being accessible during module loading, we can work around this by storing the bot info in a variable as soon as it's known.

```suggestion
let botInfo: UserFromGetMe | undefined = undefined
export default {
  fetch(request: Request, env: Environment) {
    try {
      const bot = new Bot(env.BOT_TOKEN, { botInfo });
      if (botInfo === undefined) {
        await bot.init()
        botInfo = bot.botInfo
      }
```

Please see if this type-checks, coded on GitHub.

Reply:
Done.